scRNA-seq/04-dimension_reduction_scRNA.Rmd

```diff
@@ -31,225 +44,255 @@ library(magrittr)
+`counts` with the raw data and `logcounts` with the normalized and transformed data.
+
+```{r clean_up, live = TRUE}
+assayNames(normalized_sce)
+rm(hodgkins_sce, filtered_sce)
+```
+
+
 
 ![**Status of the current dataset**](diagrams/tag-based_3.png)
 
-## Principal Components Analysis
+
+
+## Dimensionality reduction and display
+
+### Principal Components Analysis
 
 The first dimensionality reduction algorithm we will use is PCA, which we have already used!
 The idea of PCA is that it finds the axes which explain the greatest amount of variance in the data.
 
 
 ```{r PCA}
 # Run PCA
-pca_tab_mur <- calculatePCA(tm_sce_filtered)
-
-# Make a dataframe with these PCA scores
-pca_tab_mur <- data.frame(pca_tab_mur)
+hodgkins_pca <- calculatePCA(normalized_sce)
 ```
 
-```{r Plot PCA}
-# Plot this with ggplot2 and label the points with cell types
-ggplot(pca_tab_mur, 
+```{r plot_pca}
+# Plot PCA results with ggplot2 
+ggplot(data.frame(hodgkins_pca), 
        aes(x = PC1, y = PC2)) + 
-  geom_point() 
+  geom_point(alpha = 0.2) +
+  theme_bw()
 ```
-
-It looks like there are a few different clusters here, which may represent different kinds of cells within the bladder tissue.
-If we wanted to, we might look at marker genes within these clusters to try to determine which kinds of cells are in each cluster from this tissue. 
-This is exactly what the *Tabula Muris* group ended up doing, and you can see some of their results along that front at https://tabula-muris.ds.czbiohub.org/.
-
-
-## Beyond PCA: UMAP
-
-### More from the *Tabula Muris* Dataset
-
-To explore more about dimensionality reduction, we will explore a larger subset of the data from the *Tabula Muris* study. 
-We have run alevin on 15 samples, representing 9 different tissues, and taken each of those samples through QC and normalization similar to what we have presented above (with slightly different cutoffs).
-Following that, we selected the 1000 genes with the greatest biological variance (using the same method as shown in `02-normalizing_scRNA-seq.Rmd`).
-
-These results, with accompanying metadata were saved into a `SingleCellExperiment` object, which is stored as RDS file at the location `tm_normalized_file` (defined in the `filepaths` chunk above).
-Let's load that in and have a quick look at it. 
-
-```{r load_normalized}
-tm_normalized <- readr::read_rds(tm_normalized_file)
+It looks like there may be a few clusters there, though they are not very distinct.
+Sometimes looking at higher dimensions can show more (or different) structure, so lets have a look there too.
+
+```{r plot_pca2}
+# Plot a higher set of dimensions
+ggplot(data.frame(hodgkins_pca), 
+       aes(x = PC3, y = PC4)) + 
+  geom_point(alpha = 0.2) +
+  theme_bw()
 ```
 
-How many cells are in this data? How many samples are there? How many cells from each tissue?
+#### Storing PCA results with the data
```

Comment:
In terms of flow, it's not clear to me why we wouldn't use `runPCA()` first? Unless plotting using `reducedDim()` is a pain. I would think we want to encourage folks to default to saving the results as part of the object.

Reply:
This makes sense. I was going back and forth about how much to rely on the object vs. tidyverse standards, and I have been leaning more and more toward just using the object... I worry that it requires more abstract thinking and I worry that people will build odd mental models of how it is working, but the convenience level (and consistency benefits) are pretty high.